llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -676,11 +768,16 @@ Value *VPInstruction::generatePerPart(VPTransformState &State, unsigned Part) {
 
 bool VPInstruction::isVectorToScalar() const {
   return getOpcode() == VPInstruction::ExtractFromEnd ||
-         getOpcode() == VPInstruction::ComputeReductionResult;
+         getOpcode() == VPInstruction::ComputeReductionResult ||
+         getOpcode() == VPInstruction::CSAAnyActive ||
+         getOpcode() == VPInstruction::CSAAnyActiveEVL;
 }
 
 bool VPInstruction::isSingleScalar() const {
-  return getOpcode() == VPInstruction::ResumePhi;
+  return getOpcode() == VPInstruction::ResumePhi ||
+         getOpcode() == VPInstruction::CSAVLPhi ||
+         getOpcode() == VPInstruction::CSAVLSel ||
+         getOpcode() == VPInstruction::ExplicitVectorLength;
```

Comment:
Is `VPInstruction::ExplicitVectorLength` something that this patch adds? If not, how come it appears in the diff?

Reply:
I created a separate patch for this:

https://github.com/llvm/llvm-project/pull/106818